src/itest/java/uk/gov/companieshouse/company/links/steps/CompanyChargesSteps.java

```diff
@@ -0,0 +1,153 @@
+import static uk.gov.companieshouse.company.links.consumer.TestData.RESOURCE_KIND_CHARGES;
+
+import com.github.tomakehurst.wiremock.WireMockServer;
+import com.github.tomakehurst.wiremock.stubbing.ServeEvent;
+import io.cucumber.java.en.Given;
+import io.cucumber.java.en.Then;
+import io.cucumber.java.en.When;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import org.apache.commons.io.FileUtils;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.kafka.core.KafkaTemplate;
+import org.springframework.util.ResourceUtils;
+import uk.gov.companieshouse.company.links.service.CompanyProfileService;
+import uk.gov.companieshouse.stream.EventRecord;
+import uk.gov.companieshouse.stream.ResourceChangedData;
+
+public class CompanyChargesSteps {
+
+    @Value("${wiremock.server.port}")
+    private String port;
+
+    @Value("${company-links.consumer.insolvency.topic}")
+    private String topic;
+
+    private static WireMockServer wireMockServer;
+
+    private String companyNumber;
+
+    @Autowired
+    private CompanyProfileService companyProfileService;
+
+    @Autowired
+    public KafkaTemplate<String, Object> kafkaTemplate;
+
+    @Autowired
+    public KafkaConsumer<String, Object> kafkaConsumer;
+
+    @Given("Company profile stubbed with zero charges links for {string}")
+    public void company_profile_exists_without_charges(String companyNumber) {
+        this.companyNumber = companyNumber;
+        configureWiremock();
+        setGetAndPatchStubsFor(loadFileForCoNumber("profile-with-out-charges.json", companyNumber));
+     }
+
+    @When("A valid avro message is sent to the Kafka topic {string}")
+    public void send_kafka_message(String topicName) throws InterruptedException {
+        kafkaTemplate.send(topicName, createChargesMessage(companyNumber));
+        CountDownLatch countDownLatch = new CountDownLatch(1);
+        countDownLatch.await(5, TimeUnit.SECONDS);
+    }
+
+    @Then("The message is successfully consumed and company-profile-api PATCH endpoint is NOT invoked")
+    public void patchEdpointNotCalled(){
+        verify(1, getRequestedFor(urlEqualTo("/company/" + this.companyNumber + "/links")));
+        verify(0, patchRequestedFor(urlEqualTo("/company/" + this.companyNumber + "/links")));
+        wireMockServer.stop();
+    }
+
+    @Then("The message is successfully consumed and company-profile-api PATCH endpoint is invoked with charges link payload")
+    public void patchEdpointIsCalled(){
+        verify(1, getRequestedFor(urlEqualTo("/company/" + this.companyNumber + "/links")));
+        verify(1, patchRequestedFor(urlEqualTo("/company/" + this.companyNumber + "/links")));
```

Comment:
On Patch call, We should also assert that new "charges" link is added in request payload

Reply:
Does not the stub verify that because of the .withRequestBody(containing("/company/" + this.companyNumber + "/charges"))?